### DISCUSSION TITLE
Request authorization and GetStaticProps

### DISCUSSION THREAD

**Discussion by someskumar-k** [vercel/next.js#11734] [Help] [answered]
I am started to explore nextjs and I have few questions 
 1) how to get user cookies on serverside using getstaticprops?
2) does the getserversideprops will greatly impact the app perfomance?
3) Will the getstaticprops will rebuild the pages when the data is changed in database at real-time ?

**Comment by mustafa-alfar** [comment]
I implement something similar, but I encounter a problem when deploying it as static. 
here's a [StackOverflow link ](https://stackoverflow.com/questions/61142787/nextjs-cookie-return-to-initial-value-on-page-refresh)regard the problem.

**Comment by ingoclaro** [comment, marked as answer]
you won't get any cookies with getstaticprops, that method is called at build time (yarn build) and will generate a static json file. In [the docs](https://nextjs.org/docs/basic-features/data-fetching#getstaticprops-static-generation) you can see what information it gets in the context.

for 2) it depends on what your app is doing and the available resources

for 3) I don't think so, you need to run `yarn build` again to regenerate the site and then deploy it.

**Reply by someskumar-k** [reply]
Thanks for the assist, It helped @ingoclaro

**Reply by mustafa-alfar** [reply]
@ingoclaro Could we persist state with static export, 
I have tried that with cookie and it works except for static export, 
the app returns to its initial value on page refresh, 
here's the question on[ StackOverflow ](https://stackoverflow.com/questions/61142787/nextjs-cookie-return-to-initial-value-on-page-refresh)

**Reply by someskumar-k** [reply]
@mustafa-alfar , The cookie function will return cookie when the code executed at the front end(CSR) not at SSR, at client side you can  fetch cookie from req object in initial props

**Reply by itskrishnadev** [reply]
thanks for clearing out.
